### DISCUSSION TITLE
got error "has no exported member 'extendShape'."

### DISCUSSION THREAD

**Discussion by stpl-shivprasad** [pdfme/pdfme#168] [Q&A]
**Angular** : "@angular/core": "^16.0.0",
**Node** : v18.16.0

import { Component, OnInit } from '@angular/core';
import { generate, BLANK_PDF, Template } from '@pdfme/generator';

const template: Template = {
  basePdf: BLANK_PDF,
  schemas: [
    {
      a: {
        type: 'text',
        position: { x: 0, y: 0 },
        width: 10,
        height: 10,
      },
      b: {
        type: 'text',
        position: { x: 10, y: 10 },
        width: 10,
        height: 10,
      },
      c: {
        type: 'text',
        position: { x: 20, y: 20 },
        width: 10,
        height: 10,
      },
    },
  ],
};

const inputs = [{ a: 'a1', b: 'b1', c: 'c1' }];
@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
})
export class AppComponent implements OnInit {
  title = 'pdfMe';

  ngOnInit(): void {}

  downloadPdf() {
    generate({ template, inputs }).then((pdf: any) => {
      console.log(pdf);

      const blob = new Blob([pdf.buffer], { type: 'application/pdf' });
      window.open(URL.createObjectURL(blob));
    });
  }
}

**Comment by scripty08** [comment]
"zod": "3.20.2", is needed. is is installed as a subdependency of @pdfme/generator. But pdfme has a caret befor the version number. so it is installing version 3.21.3. This version seems not to export "extendShape" anymore. I have installed the 3.20.2 version and removed the caret. now it is worling.

**Reply by stpl-shivprasad** [reply]
still the same issue. even after "zod" : "3.20.2"

**Comment by stpl-shivprasad** [comment]
Hello,
Thanks for your response.

"zod" : "3.20.2" is there.  but still the same issue. I also removed &
reinstalled the package again. pls check attachment.
[image: generator compiler.png]
